src/main/java/Duke.java

```diff
@@ -1,10 +1,200 @@
+        userInput = userInput.trim();
+        String[] tokens = userInput.split(" ", 2);
+        boolean isValid = true;
+        if (tokens.length < 2) {
+            printInvalidInputMessage();
+            return;
+        }
+        switch(tokens[0]) {
+        case "done":
+            isValid = hasIndex(tokens);
+            if (isValid) {
+                markInList(Integer.parseInt(tokens[1]));
+            }
+            break;
+        case "todo":
+            addTodo(tokens[1]);
+            break;
+        case "deadline":
+            isValid = hasDeadlineOrEvent(" /by ", tokens);
+            if (isValid) {
+                int position = tokens[1].indexOf("/by");
+                addDeadline(tokens[1].substring(0, position - 1), tokens[1].substring(position + 4));
+            }
+            break;
+        case "event":
+            isValid = hasDeadlineOrEvent(" /at ", tokens);
+            if (isValid) {
+                int position = tokens[1].indexOf("/at");
+                addEvent(tokens[1].substring(0, position - 1), tokens[1].substring(position + 4));
+            }
+            break;
+        default:
+            printInvalidInputMessage();
+            break;
+        }
+        if (!isValid) {
+            printInvalidInputMessage();
+        }
+    }
+
+    public static boolean hasIndex(String[] tokens) {
+        if (tokens.length != 2) {
+            return false;
+        }
+        try {
+            Integer.parseInt(tokens[1]);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        int index = Integer.parseInt(tokens[1]);
+        return index > 0 && index <= tasksCount;
+    }
+
+    public static boolean hasDeadlineOrEvent(String keyword, String[] tokens) {
+        if (!tokens[1].contains(keyword)) {
+            return false;
+        }
+        String[] words = tokens[1].split(keyword, 2);
+        if (words.length < 2) {
+            return false;
+        }
+        String newKeyword = keyword.trim();
+        if (words[0].contains(newKeyword) || words[1].contains(newKeyword)) {
+            return false;
+        }
+        return true;
+    }
+
+
+    /** Main method */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        greet();
+        Scanner in = new Scanner(System.in);
+        String userInput = in.nextLine();
+        while (!userInput.equals("bye")) {
+            if (userInput.equals("list")) {
+                printList();
+            } else if (userInput.equals("help")) {
+                help();
+            } else {
+                processInput(userInput);
+            }
+            userInput = in.nextLine();
```

Comment:
a single line user input is processed in different functions (different levels of abstraction) [REF](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard)

Reply:
Thank you for pointing this out, I will correct it in my next increment.